src/analyses/base.ml

```diff
@@ -855,7 +854,7 @@ struct
           | [] -> assert false
           | (e1, e2) :: eqs ->
             let eqs_for_all_mem e = List.for_all (fun (e1, e2) -> CilType.Exp.(equal e1 e || equal e2 e)) eqs in
-            let eqs_map_remove e = List.map (fun (e1, e2) -> if CilType.Exp.equal e1 e then e2 else e1) eqs in
+            let eqs_map_remove e = List.of_seq @@ Seq.map (fun (e1, e2) -> if CilType.Exp.equal e1 e then e2 else e1) @@ List.to_seq eqs in
```

Comment:
What are all these `Seq` changes here? Also I'm doubtful that it does anything useful here, there's no chaining of list operations.

Reply:
@KhalilCK @montrie, could you guys take a look at this, as you mostly worked on it? Thanks!

Reply:
@mrstanb  It looks like we accidentally pushed two commits onto this branch instead of the seq-optimization branch (commits [619b634](https://github.com/goblint/analyzer/pull/943/commits/619b634ade7d40f7f6657f9dd86f5b43f7425238) and [abeffdb](https://github.com/goblint/analyzer/pull/943/commits/abeffdb83f2b8b357f8d850d249784e19022e276)) since they introduce some `Seq` notation and seem to revert changes made in your previous commit ([d7ada1c](https://github.com/goblint/analyzer/pull/943/commits/d7ada1c53e55ee58f56a2458b6d9bcc37ae678d6)).

Can we simply revert these commits or do we need to fix that by hand?

Reply:
@mrstanb @montrie 
It would be good to remove all the changes related to `Seq` from this PR to make reviewing this easier.

Reply:
Should be fixed now